app/src/main/java/com/m/androidNativeApp/MainActivity.java

```diff
@@ -42,29 +44,35 @@ protected void onCreate(Bundle savedInstanceState) {
                 .builder()
                 .build();
 
-        client.query(tasksQuery).enqueue(new ApolloCall.Callback<AllTasksQuery.Data>() {
+        Client.client.query(tasksQuery).enqueue(new ApolloCall.Callback<AllTasksQuery.Data>() {
             @Override
             public void onResponse(@NotNull Response<AllTasksQuery.Data> response) {
+
                 final AllTasksQuery.Data mResponse = response.data();
 
                 final int dataLength = mResponse.allTasks().size();
 
-                for (int i = 0; i < dataLength; i++) {
-                    taskTitle = mResponse.allTasks().get(i).fragments().taskFields().title();
-                    taskDescription = mResponse.allTasks().get(i).fragments().taskFields().description();
-                    taskId = mResponse.allTasks().get(i).fragments().taskFields().id();
-                    itemList.add(new Item(taskTitle, taskDescription, taskId));
-                }
+                MainActivity.this.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        for (int i = 0; i < dataLength; i++) {
+                            taskTitle = mResponse.allTasks().get(i).fragments().taskFields().title();
+                            taskDescription = mResponse.allTasks().get(i).fragments().taskFields().description();
+                            taskId = mResponse.allTasks().get(i).fragments().taskFields().id();
+                            itemList.add(new Item(taskTitle, taskDescription, taskId));
+                        }
```

Comment:
I don't really think this part is really needed to be performed on the main thread. In the worst case scenario, if there are many many items, this could block the whole UI thread.

I think the part to fill the `itemList` can still be done on the background thread. It's only the when you need to call `recyclerView.setAdapter`, that part needs to be done on the main ui thread.

Reply:
Thank you @wei-lee, I've updated PR.